### ISSUE TITLE
Tabs: "With Tab Icons" story has duplicated focus styles

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#63271]
## What problem does this address?

The `Tabs` ["With Tab Icons And Tooltips" story](https://wordpress.github.io/gutenberg/?path=/story/components-experimental-tabs--with-tab-icons-and-tooltips) has duplicated focus styles.

<img width="184" alt="Icon tabs with duplicate focus styles" src="https://github.com/WordPress/gutenberg/assets/555336/5d4257cc-b566-4f54-82b6-1a552278363b">

## What is your proposed solution?

Ideally, we switch the documented implementation to something that doesn't result in double focus indicators. We shouldn't add overrides to avoid this double focus, if possible.

**Comment by ciampo** [comment]
Yup, this is caused by the `<Button />` styles interfering with `Tabs`

- The "outer" outline is caused by [`<Button />`'s own focus styles](https://github.com/WordPress/gutenberg/blob/7376042df89b9f0127a1df748d03bfec957cee02/packages/components/src/button/style.scss#L36-L43)
- the inner outline is smaller than it should be because of more `<Button />` styles, [especially the icon ones](https://github.com/WordPress/gutenberg/blob/7376042df89b9f0127a1df748d03bfec957cee02/packages/components/src/button/style.scss#L299-L308), overriding the `Tab` styles

It's a tricky one. We'll probably have to add specific style overrides in the `Tabs` component to make sure that the relevant `<Button />` styles don't get applied.

Do you have any other thoughts?

**Comment by mirka** [comment]
I was hoping to compose it using existing components, without style overrides. For example, this seems to work as expected:

```jsx
<Tooltip text="WordPress">
	<Tabs.Tab tabId="tab1" aria-label="WordPress">
		<Icon icon={ wordpress } />
	</Tabs.Tab>
</Tooltip>
```

**Comment by ciampo** [comment]
That makes a lot of sense, it's so good to start seeing the results of composing smaller, modular components!

I implemented your suggestion in #63297
